index.js

```diff
@@ -95,20 +147,25 @@ function testGenResponse(swagger, path, operation, response, config) {
  * @param  {string} path API path to generate tests for
  * @param  {string} operation operation of the path to generate tests for
  * @param  {json} config configuration for testGen
+ * @param  {json} parameters json for different parameters
  * @returns {string|Array} set of all tests for a path's operation
  */
-function testGenOperation(swagger, path, operation, config) {
+function testGenOperation(swagger, path, operation, config, parameters) {
   var responses = swagger.paths[path][operation].responses;
   var result = [];
   var res;
+  var origin = JSON.parse(JSON.stringify(parameters));
 
   for (res in responses) {
     if (responses.hasOwnProperty(res)) {
-      result.push(testGenResponse(swagger,
-        path,
-        operation,
-        res,
-        config));
+      parameters = JSON.parse(JSON.stringify(origin));
+      if (responses.hasOwnProperty(res)) {
+        result.push(testGenResponse(swagger,
+          path,
+          operation,
+          res,
+          config, parameters));
```

Comment:
I think I made it this way b/c it has length more than 80 characters, while other don't. I
'll change it all new line for each parameter, cause this one cannot be put in one line